example/tostring_jsonenums.go

```diff
@@ -0,0 +1,93 @@
+// generated by jsonenums -stringer=true -type=toString; DO NOT EDIT
+
+package main
+
+import (
+	"database/sql/driver"
+	"encoding/json"
+	"fmt"
+)
+
+var (
+	_toStringNameToValue = map[string]toString{
+		"elemA": elemA,
+		"elemB": elemB,
+		"elemC": elemC,
+	}
+
+	_toStringValueToName = map[toString]string{
+		elemA: "elemA",
+		elemB: "elemB",
+		elemC: "elemC",
+	}
+)
+
+func init() {
+	var v toString
+	if _, ok := interface{}(v).(fmt.Stringer); ok {
+		_toStringNameToValue = map[string]toString{
+			interface{}(elemA).(fmt.Stringer).String(): elemA,
+			interface{}(elemB).(fmt.Stringer).String(): elemB,
+			interface{}(elemC).(fmt.Stringer).String(): elemC,
+		}
+	}
+}
+
+func (r toString) String() string {
+	return _toStringValueToName[r]
```

Comment:
One idea is to provide a method other than `String()` and let the consumer decide if they want the behavior they get from this to be what satisfies `Stringer`.